### PR TITLE
loader: return error if parse mydumper meta data failed

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
fix issue: https://internal.pingcap.net/jira/browse/TOOL-1425
should return error in load unit if parse mydumper metadata failed, otherwise will get error "invalid mysql position string: " in sync unit, this will make user confused.

### What is changed and how it works?
handle the error for `getMydumpMetadata`

### CODE REVIEW

File: loader/loader.go [hunk 1]
```diff
@@ -421,7 +421,12 @@ func (l *Loader) Process(ctx context.Context, pr chan pb.ProcessResult) {
 	defer cancel()
 
 	l.newFileJobQueue()
-	l.getMydumpMetadata()
+	if err := l.getMydumpMetadata(); err != nil {
+		pr <- pb.ProcessResult{
```
Comment (amyangfei) [inline loader/loader.go]: `loaderExitWithErrorCounter.WithLabelValues(l.cfg.Name).Inc()`
increase `loaderExitWithErrorCounter` here?

File: loader/loader.go [hunk 2]
```diff
@@ -421,7 +421,12 @@ func (l *Loader) Process(ctx context.Context, pr chan pb.ProcessResult) {
 	defer cancel()
 
 	l.newFileJobQueue()
-	l.getMydumpMetadata()
+	if err := l.getMydumpMetadata(); err != nil {
```
Comment (amyangfei) [inline loader/loader.go]: `loaderExitWithErrorCounter.WithLabelValues(l.cfg.Name).Inc()`
increase `loaderExitWithErrorCounter` in this code block?

Comment (WangXiangUSTC) [reply loader/loader.go]: updated

**Pull request merged**